ingestion/ingestion_server/ingestion_file_upload_rest_api.py

```diff
@@ -332,3 +347,56 @@ def file_upload_main_cors(file_path):
     :return: Static file from directory
     """
     return send_from_directory("file_upload/cors", file_path)
+
+
+@ingestion_file_upload_bp.route('/file_share/', methods=["GET"])
+def file_share():
+    """
+    URL handler needed for the jquery-file-share integration.
+    :param file_path:  Maps URL to files in templates directory.
+    :return: Static file from directory along with data to display
+    """
+    # TODO: Display all the files of the logged in owner only
+    response = AssetDbApi.get_all()
+    return render_template('share.html', data=response)
+
+
+@ingestion_file_upload_bp.route('/file_share/', methods=["POST"])
+def file_sharing():
+    """
+    :return:
+    """
+    success, documents, msg = AssetDbApi.get_asset(request.form["file"])
+    asset_id = documents[0]['uuid']
+    file_id = documents[0]['grid_fid']
+    try:
+        server_urls_instance = ServerUrls().get_instance()
+        get_return_obj = RestClientApis.http_get_and_check_success(
+            server_urls_instance.key_server_asset_url+asset_id+"/")
+        if get_return_obj.success:
+            symmetric_key = get_return_obj.to_dict()['json']['response']['key']['key']
+            db_core = MainDb.get_core_db_instance()
+            fs = gridfs.GridFSBucket(db_core.get_magen_mdb())
+            files_data = fs.find({'_id': file_id})
+            for files in files_data:
+                user_pubkey = fs.find({'metadata.owner': files.metadata['owner'], 'metadata.type': 'public key'})
+            if user_pubkey:
+                for name in user_pubkey:
+                    fname = name.filename
+                src_file_path = os.path.join(IngestionGlobals().data_dir, fname)
+                try:
+                    with open(src_file_path, 'rb') as f:
+                        data = f.read()
+                    iv = Random.new().read(AES.block_size)
+                    cipher = AES.new(data, AES.MODE_CBC, iv)
+                    cipher_text = cipher.encrypt(symmetric_key)
```

Comment:
I guess you will be working on the part where the encrypted key is put under a dynamic URL so it can be shared with the user